nodeadm/cmd/nodeadm/init/init.go

```diff
@@ -168,6 +168,15 @@ func enrichConfig(log *zap.Logger, cfg *api.NodeConfig) error {
 	if err != nil {
 		return err
 	}
+	if awsConfig.RetryMaxAttempts == 0 {
+		// use a very generous retry policy to accomodate delays in network readiness
+		// we only specify the max attempts if it is unset by the user
+		// so it's possible to override with the AWS_MAX_ATTEMPTS environment variable.
+		// NOTE that this is the number of attempts that will be made in a blocking fashion
+		// i.e. an SDK client.ExampleAPICall() will not return until these attempts are exhausted
+		// we'll give up after approximately 10 minutes
+		awsConfig.RetryMaxAttempts = 30
```

Comment:
The math for this is:
1. Max delay between attempts is 20 seconds.
2. Delay is exponential, starting at 2; but there's jitter applied. So, the actual delay is 0-100% of the quantity.

The sequence looks like:
| Attempt | Delay (s) |
|--------|--------|
| 1 | 2 |
| 2 | 4 |
| 3 | 8 |
| 4 | 16 |
| 5 | 20 | 

Any attempt after the fifth attempt will have a delay of 20 seconds. Jitter is not applied once the max delay has been reached. We'll just treat each attempt as taking 20 seconds, the first 4 attempts don't change much.

So, 10 minutes (600 seconds) / 20 seconds = 30 attempts.

Reply:
Note that if the connection times out, the delay between each attempt will be 30 seconds longer. That's where our custom waiter's timeout comes in -- we will still give up after about 10 minutes, even if we don't make all 30 attempts.